packages/sycamore-macro/src/view/parse.rs

```diff
@@ -175,43 +177,49 @@ impl Parse for AttributeType {
             }
         }
 
-        let ident: AttributeName = input.parse()?;
-        let name = ident.to_string();
-
-        if name == "ref" {
-            Ok(Self::Ref)
-        } else if name == "dangerously_set_inner_html" {
-            Ok(Self::DangerouslySetInnerHtml)
-        } else if input.peek(Token![:]) {
-            let _colon: Token![:] = input.parse()?;
-            match name.as_str() {
-                "on" => {
-                    let event = input.call(Ident::parse_any)?;
-                    Ok(Self::Event {
-                        event: event.to_string(),
-                    })
-                }
-                "prop" => {
-                    let prop = input.call(Ident::parse_any)?;
-                    Ok(Self::Property {
-                        prop: prop.to_string(),
-                    })
-                }
-                "bind" => {
-                    let prop = input.call(Ident::parse_any)?;
-                    Ok(Self::Bind {
-                        prop: prop.to_string(),
-                    })
+        let lookahead = input.lookahead1();
+        if lookahead.peek(Token![..]) {
+            let _2dot = input.parse::<Token![..]>()?;
+            Ok(Self::Spread)
+        } else {
```

Comment:
For better error messages, we should also have `else if lookahead.peek(Ident)` and an `else { Err(lookahead.error()) }`

Reply:
Actually the error message will be perfectly fine, it's the same one it was before. It'll just say "expected Ident, found {blank}".

Reply:
I literally just put the old code in the `else` statement

Reply:
That's a fair point. I'll look into it once I have the event stuff reimplemented.

Reply:
the `if lookahead.peek(Ident)` version doesn't work because `Ident::parse` includes keywords but `peek(Ident)` does not, which fails on `ref`. I instead used `map_err` on the name parse to show the improved error.

Reply:
Sorry for taking so long to respond. It is possible to do use [`Ident::peek_any`](https://docs.rs/syn/latest/syn/ext/trait.IdentExt.html#associatedconstant.peek_any) which has the analogous behavior to `Ident::parse_any`

Reply:
No worries, I was able to use my time to implement this change on the headless-ui port I mainly designed it for. I can do the peek change if you want, I figured that out after doing the error map but didn't think it was a big enough difference to warrant a commit. I'll get it done now real quick.